// @wordpress/block-library/build-module/post-featured-image/dimension-controls.js
import { createElement, Fragment } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';
import { PanelBody, __experimentalUnitControl as UnitControl, BaseControl, Flex, FlexItem, __experimentalSegmentedControl as SegmentedControl, __experimentalSegmentedControlOption as SegmentedControlOption, __experimentalUseCustomUnits as useCustomUnits } from '@wordpress/components';
import { useSetting } from '@wordpress/block-editor';
const SCALE_OPTIONS = createElement(Fragment, null, createElement(SegmentedControlOption, {
  value: "cover",
  label: _x('Cover', 'Scale option for Image dimension control')
}), createElement(SegmentedControlOption, {
  value: "contain",
  label: _x('Contain', 'Scale option for Image dimension control')
}), createElement(SegmentedControlOption, {
  value: "fill",
  label: _x('Stretch', 'Scale option for Image dimension control')
}));

const DimensionControls = ({
  attributes: {
    width,
    height,
    scale
  },
  setAttributes
}) => {
  const defaultUnits = ['px', '%', 'vw', 'em', 'rem'];
  const units = useCustomUnits({
    availableUnits: useSetting('spacing.units') || defaultUnits
  });

  const onDimensionChange = (dimension, nextValue) => {
    const parsedValue = parseFloat(nextValue);
    /**
     * If we have no value set and we change the unit,
     * we don't want to set the attribute, as it would
     * end up having the unit as value without any number.
     */

    if (isNaN(parsedValue) && nextValue) return;
    setAttributes({
      [dimension]: parsedValue < 0 ? '0' : nextValue
    });
  };

  const scaleLabel = _x('Scale', 'Image scaling options');

  return createElement(PanelBody, {
    title: __('Dimensions')
  }, createElement(Flex, {
    justify: "space-between"
  }, createElement(FlexItem, null, createElement(UnitControl, {
    label: __('Height'),
    labelPosition: "top",
    value: height || '',
    onChange: nextHeight => {
      onDimensionChange('height', nextHeight);
    },
    units: units
  })), createElement(FlexItem, null, createElement(UnitControl, {
    label: __('Width'),
    labelPosition: "top",
    value: width || '',
    onChange: nextWidth => {
      onDimensionChange('width', nextWidth);
    },
    units: units
  }))), !!height && createElement(Fragment, null, createElement(BaseControl, {
    "aria-label": scaleLabel,
    className: "block-library-post-featured-image-scale-controls"
  }, createElement("div", null, createElement(BaseControl.VisualLabel, null, scaleLabel)), createElement(SegmentedControl, {
    label: scaleLabel,
    value: scale,
    onChange: value => {
      setAttributes({
        scale: value
      });
    },
    isBlock: true
  }, SCALE_OPTIONS))));
};

export default DimensionControls;
