// @wordpress/block-library/build/embed/edit.native.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _util = require("./util");

var _embedControls = _interopRequireDefault(require("./embed-controls"));

var _icons = require("./icons");

var _embedLoading = _interopRequireDefault(require("./embed-loading"));

var _embedPlaceholder = _interopRequireDefault(require("./embed-placeholder"));

var _embedPreview = _interopRequireDefault(require("./embed-preview"));

var _embedBottomSheet = _interopRequireDefault(require("./embed-bottom-sheet"));

var _classnames = _interopRequireDefault(require("classnames"));

var _i18n = require("@wordpress/i18n");

var _data = require("@wordpress/data");

var _blockEditor = require("@wordpress/block-editor");

var _coreData = require("@wordpress/core-data");

var _primitives = require("@wordpress/primitives");

/**
 * Internal dependencies
 */

/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
const EmbedEdit = props => {
  const {
    attributes: {
      providerNameSlug,
      previewable,
      responsive,
      url
    },
    attributes,
    isSelected,
    onReplace,
    setAttributes,
    insertBlocksAfter,
    onFocus,
    clientId
  } = props;
  const defaultEmbedInfo = {
    title: (0, _i18n._x)('Embed', 'block title'),
    icon: _icons.embedContentIcon
  };
  const {
    icon,
    title
  } = (0, _util.getEmbedInfoByProvider)(providerNameSlug) || defaultEmbedInfo;
  const {
    wasBlockJustInserted
  } = (0, _data.useSelect)(select => ({
    wasBlockJustInserted: select(_blockEditor.store).wasBlockJustInserted(clientId, 'inserter_menu')
  }), [clientId]);
  const [isEditingURL, setIsEditingURL] = (0, _element.useState)(isSelected && wasBlockJustInserted && !url);
  const {
    preview,
    fetching,
    cannotEmbed
  } = (0, _data.useSelect)(select => {
    const {
      getEmbedPreview,
      isPreviewEmbedFallback,
      isRequestingEmbedPreview
    } = select(_coreData.store);

    if (!url) {
      return {
        fetching: false,
        cannotEmbed: false
      };
    }

    const embedPreview = getEmbedPreview(url);
    const previewIsFallback = isPreviewEmbedFallback(url); // The external oEmbed provider does not exist. We got no type info and no html.

    const badEmbedProvider = (embedPreview === null || embedPreview === void 0 ? void 0 : embedPreview.html) === false && (embedPreview === null || embedPreview === void 0 ? void 0 : embedPreview.type) === undefined; // Some WordPress URLs that can't be embedded will cause the API to return
    // a valid JSON response with no HTML and `code` set to 404, rather
    // than generating a fallback response as other embeds do.

    const wordpressCantEmbed = (embedPreview === null || embedPreview === void 0 ? void 0 : embedPreview.code) === '404';
    const validPreview = !!embedPreview && !badEmbedProvider && !wordpressCantEmbed; // `isRequestingEmbedPreview` is returning false just before an
    // `apiFetch` is triggered. We're assuming that a fetch is happening
    // if there is an `attributesUrl` set but there is no data in
    // `embedPreview` which represents the response returned from the API.

    const isFetching = isRequestingEmbedPreview(url) || url && !embedPreview;
    return {
      preview: validPreview ? embedPreview : undefined,
      fetching: isFetching,
      cannotEmbed: !validPreview || previewIsFallback
    };
  }, [url]);
  /**
   * @return {Object} Attributes derived from the preview, merged with the current attributes.
   */

  const getMergedAttributes = () => {
    const {
      allowResponsive,
      className
    } = attributes;
    return { ...attributes,
      ...(0, _util.getAttributesFromPreview)(preview, title, className, responsive, allowResponsive)
    };
  };

  (0, _element.useEffect)(() => {
    if (!(preview !== null && preview !== void 0 && preview.html) || !cannotEmbed || fetching) {
      return;
    } // At this stage, we're not fetching the preview and know it can't be embedded,
    // so try removing any trailing slash, and resubmit.


    const newURL = url.replace(/\/$/, '');
    setIsEditingURL(false);
    setAttributes({
      url: newURL
    });
  }, [preview === null || preview === void 0 ? void 0 : preview.html, url]); // Handle incoming preview

  (0, _element.useEffect)(() => {
    if (preview && !isEditingURL) {
      // Even though we set attributes that get derived from the preview,
      // we don't access them directly because for the initial render,
      // the `setAttributes` call will not have taken effect. If we're
      // rendering responsive content, setting the responsive classes
      // after the preview has been rendered can result in unwanted
      // clipping or scrollbars. The `getAttributesFromPreview` function
      // that `getMergedAttributes` uses is memoized so that we're not
      // calculating them on every render.
      setAttributes(getMergedAttributes());

      if (onReplace) {
        const upgradedBlock = (0, _util.createUpgradedEmbedBlock)(props, getMergedAttributes());

        if (upgradedBlock) {
          onReplace(upgradedBlock);
        }
      }
    }
  }, [preview, isEditingURL]);
  const blockProps = (0, _blockEditor.useBlockProps)();

  if (fetching) {
    return (0, _element.createElement)(_primitives.View, blockProps, (0, _element.createElement)(_embedLoading.default, null));
  }

  const showEmbedPlaceholder = !preview || cannotEmbed;
  const {
    type,
    className: classFromPreview
  } = getMergedAttributes();
  const className = (0, _classnames.default)(classFromPreview, props.className);
  return (0, _element.createElement)(_element.Fragment, null, showEmbedPlaceholder ? (0, _element.createElement)(_primitives.View, blockProps, (0, _element.createElement)(_embedPlaceholder.default, {
    icon: icon,
    isSelected: isSelected,
    label: title,
    onPress: event => {
      onFocus(event);
      setIsEditingURL(true);
    },
    cannotEmbed: cannotEmbed
  })) : (0, _element.createElement)(_element.Fragment, null, (0, _element.createElement)(_embedControls.default, {
    showEditButton: preview && !cannotEmbed,
    switchBackToURLInput: () => setIsEditingURL(true)
  }), (0, _element.createElement)(_primitives.View, blockProps, (0, _element.createElement)(_embedPreview.default, {
    className: className,
    clientId: clientId,
    icon: icon,
    insertBlocksAfter: insertBlocksAfter,
    isSelected: isSelected,
    label: title,
    onFocus: onFocus,
    preview: preview,
    previewable: previewable,
    type: type,
    url: url
  }))), (0, _element.createElement)(_embedBottomSheet.default, {
    value: url,
    isVisible: isEditingURL,
    onClose: () => setIsEditingURL(false),
    onSubmit: value => {
      setIsEditingURL(false);
      setAttributes({
        url: value
      });
    }
  }));
};

var _default = EmbedEdit;
exports.default = _default;
